nerfstudio/utils/tensor_dataclass.py

```diff
@@ -141,6 +141,9 @@ def _broadcast_dict_fields(self, dict_: Dict, batch_shape) -> Dict:
                 new_dict[k] = v.broadcast_to(batch_shape)
             elif isinstance(v, Dict):
                 new_dict[k] = self._broadcast_dict_fields(v, batch_shape)
+            else:
+                # Don't broadcast the remaining fields
+                new_dict[k] = v
```

Comment:
Why is this changed?

Reply:
This comes from the support for the fisheye mask radius parameter in the `metadata` dict that I added. I discussed it a bit in the discussion in the PR, quoted below:

> I'm not super happy with this solution since it doesn't show up in the config output in the command line, and requires updating the _broadcast_dict_fields methods to not overwrite metadata, whose implications I don't understand.

In slightly more detail - it's because when creating the `Cameras` object (https://github.com/nerfstudio-project/nerfstudio/blob/main/nerfstudio/data/dataparsers/nerfstudio_dataparser.py#L292-L303), any populated `metadata` seems to get dropped due to the call through this function. Specifically, we end up calling this `_broadcast_dict_fields` method with `dict_ = metadata, batch_shape = batch_shape`, and then iterating through all the keys of the metadata. Since the one key that's currently set (https://github.com/nerfstudio-project/nerfstudio/blob/main/nerfstudio/data/dataparsers/nerfstudio_dataparser.py#L291) is not a `torch.Tensor`, `TensorDataClass`, or `Dict`, it silently falls through the if statement and gets dropped from the new output dictionary. The created `Cameras` object then gets a blank `metadata` dict instead of one containing `fisheye_crop_radius` (and any other metadata keys that might be added in the future).

Reply:
Okay got it. I think this is fine and won't have downstream implications. I see there was already support for metadata and "fisheye_crop_radius", so why wasn't this needed before for the base_datamanager.py Cameras inititialization?

Reply:
Prior to this, the `metadata` in `Cameras` was being overwritten to `{}`, which I think was a bug in the implementation. I think the reason it didn't get caught is because the value is (probably?) normally specified as a command line parameter, which takes a different path to populate the fisheye mask radius and doesn't exercise this code.

Reply:
I also fixed this here: https://github.com/nerfstudio-project/nerfstudio/pull/2783 (which includes a test)